ws-client/src/jsonrpc_transport.rs

```diff
@@ -3,11 +3,12 @@
 //! Wraps the underlying WebSocket transport with specific JSONRPC details.
 
 use crate::transport::{self, WsConnectError, WsNewDnsError};
+use crate::WsConfig;
 use jsonrpsee_types::jsonrpc;
 
 /// Creates a new JSONRPC WebSocket connection, represented as a Sender and Receiver pair.
-pub async fn websocket_connection(target: impl AsRef<str>) -> Result<(Sender, Receiver), WsNewDnsError> {
-	let (sender, receiver) = transport::websocket_connection(target.as_ref()).await?;
+pub async fn websocket_connection<'a>(config: WsConfig) -> Result<(Sender, Receiver), WsNewDnsError> {
```

Comment:
Why was the liftetime needed here?

Reply:
It was a leftover/nit when I experimented with `&'a str` in the WebSocketConfig but I will cleanup the code and let you know when it's ready for another round of reviews.